cfg/Parameters.cfg

```diff
@@ -40,9 +40,9 @@ PACKAGE='control_toolbox'
 
 def generate(gen):
 	#        Name            Type    Level  Description      Default  Min  Max
-	gen.add( "p" ,      double_t, 1,"Proportional gain.", 10.0 , 0 , 100000)
-	gen.add( "i" ,      double_t, 1,"Integral gain.",     0.1 , 0 , 1000)
-	gen.add( "d" ,      double_t, 1,"Derivative gain.",   1.0 , 0 , 1000)
+	gen.add( "p" ,      double_t, 1,"Proportional gain.", 10.0 , -100000 , 100000)
```

Comment:
Just wondering if it makes sense to have a negative proportional gain wrt to the error.
I'd say the same for the integral gain.
AFAIK, only the derivative gain can be negative in a stable PID controller.

Maybe you're computing your error as actual - desired (setpoint) in your controller?!

If there's a case where negative values for the proportional and integral gains makes sense, could you provide it?
For the derivative the condition for a stable PID controller could still allow for negative values though.

http://www.mathworks.com/help/slcontrol/ug/troubleshooting-automatic-pid-tuning.html